### PR TITLE
fix: add examples paramStructure by-name support

### LINKED ISSUE #26: support paramStructure by-name

### LINKED ISSUE #309: Outdated example?
Using code from [README](/open-rpc/docs-react/blob/master/README.md) leads to an error:

```
TypeScript error in index.tsx(4,10):
Module '"../node_modules/@open-rpc/meta-schema/build/src"' has no exported member 'OpenRPC'. Did you mean 'Openrpc'?  TS2724
```
It seems it should be something like this

```typescript
import React from 'react';
import ReactDOM from 'react-dom';
import Documentation from "@open-rpc/docs-react";
import { OpenrpcDocument } from '@open-rpc/meta-schema';

const schema: OpenrpcDocument = {
    openrpc: "1.0.0",
    info: {
        "version": "0.0.0-development",
        "title": "My New API"
    },
    methods: []
};

ReactDOM.render(<Documentation schema={schema} />, document.getElementById("root"));
```

### LINKED ISSUE #320: Support for arguments by name in examples
After playing arround for a while I found that, when defining a call's example as a Example Pairing Object, the params for this MUST be inside a list, and I think It does not cover the case of calls where all params are defined by name. For example, for the given definition:
```json
{
  "openrpc": "1.0.0-rc1",
  "info": {
    "version": "1.0.0",
    "title": "Petstore",
    "license": {
      "name": "MIT"
    }
  },
  "servers": [
    {
      "url": "http://localhost:8080"
    }
  ],
  "methods": [
    {
      "name": "get_pet",
      "summary": "Info for a specific pet",
      "paramStructure": "by-name",
      "tags": [
        {
          "name": "pets"
        }
      ],
      "params": [
        {
          "name": "petId",
          "required": true,
          "description": "The id of the pet to retrieve",
          "schema": {
            "type": "object"
          }
        }
      ],
      "result": {
        "name": "pet",
        "description": "Expected response to a valid request",
        "schema": {
          "$ref": "#/components/schemas/Pet"
        }
      },
      "examples": [
          {"name": "get_pet_example",
          "params": [
              {
                  "name": "pet_id",
                  "value": 1
              }
          ],
          "result":{
              "name": "get_pet_result",
              "value": {
                  "id": 1,
                  "name": "Vivian" 
              }
          }
        }
      ]
    }
  ],
  "components": {
    "schemas": {
      "Pet": {
        "type": "object",
        "required": [
          "id",
          "name"
        ],
        "properties": {
          "id": {
            "type": "integer"
          },
          "name": {
            "type": "string"
          },
          "tag": {
            "type": "string"
          }
        }
      },
      "Pets": {
        "type": "array",
        "items": {
          "$ref": "#/components/schemas/Pet"
        }
      }
    }
  }
}
```

The resulting example display in the Open-RPC playground shows:
image

and I think the correct request should be of the form
```json
{
"id":1,
"jsonrpc":"2.0",
"method":"get_pet",
"params":{
"petID":1
}
}
```

From my point of view, 'params' field in Example Pairing Object should be allowed being a single Example object. Am I Wrong?

### PR DESCRIPTION
**Pull request by shanejonas**
fixes #320
fixes #309
fixes #26

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,10 +44,10 @@ npm install @open-rpc/docs-react @open-rpc/meta-schema --save
 import React from 'react';
 import ReactDOM from 'react-dom';
 import Documentation from "@open-rpc/docs-react";
-import { OpenRPC } from '@open-rpc/meta-schema';
+import { OpenRpcDocument } from '@open-rpc/meta-schema';
```
Comment (meowsbits) [inline README.md:47]: Capitalize RPC?

Comment (shanejonas) [reply README.md:47]: take it up with `@open-rpc/meta-schema`

Comment (meowsbits) [reply README.md:47]: Ah, nm, generated code.

File: src/ExamplePairing/ExamplePairing.tsx [hunk 2]
```diff
@@ -6,6 +6,8 @@ import ReactMarkdown from "react-markdown";
 import { MethodObject, ExampleObject, ExamplePairingObject } from "@open-rpc/meta-schema";
 import _ from "lodash";
 
+export type TParamStructure = "either" | "by-name" | "by-position";
```
Comment (meowsbits) [inline src/ExamplePairing/ExamplePairing.tsx:9]: I'm assuming type `string` is inferred here?

File: README.md (hunk 1 above)
Comment (meowsbits) [reply README.md:47]: lol quick res, didn't even load ya before my follow up ;)
yeap.

Comment (shanejonas) [inline README.md:47]:
```suggestion
import { OpenrpcDocument } from '@open-rpc/meta-schema';
```

**Comment by shanejonas** [comment]
needs a test or two

File: src/ExamplePairing/ExamplePairing.tsx (hunk 2 above)
Comment (BelfordZ) [reply src/ExamplePairing/ExamplePairing.tsx:9]: good q.

Comment (shanejonas) [reply src/ExamplePairing/ExamplePairing.tsx:9]: yep

Comment (meowsbits) [reply src/ExamplePairing/ExamplePairing.tsx:9]: affirmative

**Comment by shanejonas** [comment]
👍

File: src/ExamplePairing/ExamplePairing.tsx (hunk 2 above)
Comment (BelfordZ) [inline src/ExamplePairing/ExamplePairing.tsx:9]: this should be importable from the meta-schema, but its k for now

File: src/ExamplePairing/ExamplePairing.tsx [hunk 3]
```diff
@@ -31,10 +33,18 @@ class ExamplePairing extends Component<IProps, {}> {
     if (!example || _.isEmpty(example)) {
       return null;
     }
+    const paramStructure: TParamStructure = method?.paramStructure || "either";
```
Comment (BelfordZ) [inline src/ExamplePairing/ExamplePairing.tsx:36]: does typescript produce something that works in browsers for this?

`?` isnt all that well supported yet.

File: src/ExamplePairing/ExamplePairing.tsx [hunk 4]
```diff
@@ -31,10 +33,18 @@ class ExamplePairing extends Component<IProps, {}> {
...
+    const params = paramStructure === "by-name"
```
Comment (BelfordZ) [inline src/ExamplePairing/ExamplePairing.tsx:37]: maybe ternary isnt the best option here

File: src/ExamplePairing/ExamplePairing.tsx (hunk 3 above)
Comment (shanejonas) [reply src/ExamplePairing/ExamplePairing.tsx:36]: yea you just `target: es5`

**Pull request merged**